src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -156,14 +156,14 @@ public AddOnInfoAndVersions getByUid(String uid) throws IOException {
 	}
 	
 	@Override
-	public Collection<AddOnInfoAndVersions> getByTag(String tag) throws Exception {
+	public Collection<AddOnInfoSummary> getByTag(String tag) throws Exception {
 		SearchResult result = client.execute(new Search.Builder(new SearchSourceBuilder()
 				.size(SEARCH_SIZE)
 				.query(QueryBuilders.matchQuery("tags", tag)).toString())
 				.addIndex(AddOnInfoAndVersions.ES_INDEX)
 				.build());
 		return result.getHits(AddOnInfoAndVersions.class).stream()
-				.map(sr -> sr.source)
+				.map(sr -> new AddOnInfoSummary(sr.source))
```

Comment:
Yes, just leave this as it was. There's no reason to discard information pre-emptively on the back-end. IIRC the purpose of the AddOnInfoAndSummary class is so we can send less information across the network to the REST client, but when we're on the server we want to keep all the info.